### ISSUE TITLE
osu!lazer does crash after finishing a beatmap

### ISSUE THREAD

**Issue by Jasonotto** [ppy/osu#1324]
osu!lazer does crash after finishing a beatmap

logs:
[crashonfinish.zip](https://github.com/ppy/osu/files/1337576/crashonfinish.zip)

**Comment by peppy** [comment]
```27.09.2017 12:51:31: ERROR: Could not import beatmap set
27.09.2017 12:51:31: System.FormatException: Die Eingabezeichenfolge hat das falsche Format.
27.09.2017 12:51:31: bei System.Number.ParseDouble(String value, NumberStyles options, NumberFormatInfo numfmt)
27.09.2017 12:51:31: bei osu.Game.Beatmaps.Formats.OsuLegacyDecoder.handleTimingPoints(Beatmap beatmap, String line)
27.09.2017 12:51:31: bei osu.Game.Beatmaps.Formats.OsuLegacyDecoder.ParseFile(StreamReader stream, Beatmap beatmap)
27.09.2017 12:51:31: bei osu.Game.Beatmaps.Formats.BeatmapDecoder.ParseFile(StreamReader stream)
27.09.2017 12:51:31: bei osu.Game.Beatmaps.Formats.OsuLegacyDecoder.ParseFile(StreamReader stream)
27.09.2017 12:51:31: bei osu.Game.Beatmaps.Formats.OsuLegacyDecoder.Decode(StreamReader stream)
27.09.2017 12:51:31: bei osu.Game.Beatmaps.BeatmapManager.importToStorage(ArchiveReader reader)
27.09.2017 12:51:31: bei osu.Game.Beatmaps.BeatmapManager.<>c__DisplayClass35_0.<Import>b__0()
27.09.2017 12:51:31: bei SQLite.Net.SQLiteConnection.RunInTransaction(Action action)
27.09.2017 12:51:31: bei osu.Game.Beatmaps.BeatmapManager.Import(ArchiveReader archiveReader)
27.09.2017 12:51:31: bei osu.Game.Beatmaps.BeatmapManager.Import(String[] paths)
27.09.2017 12:52:06: ERROR: Could not import beatmap set
27.09.2017 12:52:06: System.InvalidOperationException: No beatmap files found in the map folder.
27.09.2017 12:52:06: bei osu.Game.Beatmaps.BeatmapManager.importToStorage(ArchiveReader reader)
27.09.2017 12:52:06: bei osu.Game.Beatmaps.BeatmapManager.<>c__DisplayClass35_0.<Import>b__0()
27.09.2017 12:52:06: bei SQLite.Net.SQLiteConnection.RunInTransaction(Action action)
27.09.2017 12:52:06: bei osu.Game.Beatmaps.BeatmapManager.Import(ArchiveReader archiveReader)
27.09.2017 12:52:06: bei osu.Game.Beatmaps.BeatmapManager.Import(String[] paths)```

Can you provide a link to the beatmap responsible for the crash? Looks to be a parsing issue.

**Comment by smoogipoo** [comment]
In addition to the above:

```
27.09.2017 14:27:33: bei osu.Game.Screens.Ranking.ResultsPageScore.DrawableScoreStatistic.load(OsuColour colours)
27.09.2017 14:27:33: --- Ende der internen Ausnahmestapelüberwachung ---
27.09.2017 14:27:33: bei System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
27.09.2017 14:27:33: bei System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
27.09.2017 14:27:33: bei System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
27.09.2017 14:27:33: bei osu.Framework.Allocation.DependencyContainer.<>c__DisplayClass7_1.<register>b__5(Object instance)
27.09.2017 14:27:33: bei System.Collections.Generic.List`1.ForEach(Action`1 action)
27.09.2017 14:27:33: bei osu.Framework.Allocation.DependencyContainer.<>c__DisplayClass7_2.<register>b__7(DependencyContainer container, Object instance)
27.09.2017 14:27:33: bei osu.Framework.Allocation.DependencyContainer.Inject[T](T instance, Boolean autoRegister, Boolean lazy)
27.09.2017 14:27:33: bei osu.Framework.Graphics.Drawable.Load(IFrameBasedClock clock, IReadOnlyDependencyContainer dependencies)
27.09.2017 14:27:33: bei osu.Framework.Graphics.Containers.CompositeDrawable.loadChild(Drawable child)
27.09.2017 14:27:33: bei osu.Framework.Graphics.Containers.CompositeDrawable.AddInternal(Drawable drawable)
27.09.2017 14:27:33: bei osu.Framework.Graphics.Containers.Container`1.AddInternal(Drawable drawable)
27.09.2017 14:27:33: bei osu.Framework.Graphics.Containers.Container`1.AddRange(IEnumerable`1 range)
27.09.2017 14:27:33: bei osu.Game.Screens.Ranking.ResultsPageScore.load(OsuColour colours)
27.09.2017 14:27:33: System.IO.FileNotFoundException: Die Datei oder Assembly "Microsoft.CSharp, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden.
```

Looks like your .NET Framework installation is corrupt (or you don't have the correct one installed). Install https://www.microsoft.com/en-us/download/details.aspx?id=49981 and if that doesn't work, run the .NET Framework repair tool (https://www.microsoft.com/en-us/download/details.aspx?id=30135).

**Comment by Jasonotto** [comment]
it does actually crash on every beatmap

**Comment by peppy** [comment]
This is not related to .NET installation

**Comment by Jasonotto** [comment]
> Can you provide a link to the beatmap responsible for the crash? Looks to be a parsing issue.

it was an empty beatmap

> Looks like your .NET Framework installation is corrupt (or you don't have the correct one installed). Install https://www.microsoft.com/en-us/download/details.aspx?id=49981 and if that doesn't work, run the .NET Framework repair tool (https://www.microsoft.com/en-us/download/details.aspx?id=30135).

didnt work :/

**Comment by peppy** [comment]
One of your issues was fixed by #1325.

If you are compiling manually, then please try again on the lastest master branch.

As for the parsing issue, it would be useful to find the precise line which is cuausing the problem via debugging, if you're able to do that.

**Comment by peppy** [comment]
Closing as it's probably fixed.
